azuredevops/internal/service/approvalsandchecks/resource_check_approval.go

```diff
@@ -0,0 +1,129 @@
+package approvalsandchecks
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/utils/pipelineschecksextras"
+)
+
+// ResourceCheckApproval schema and implementation for branch check resources
+func ResourceCheckApproval() *schema.Resource {
+	r := genBaseCheckResource(flattenCheckApproval, expandCheckApproval)
+
+	r.Schema["approvers"] = &schema.Schema{
+		Type:     schema.TypeList,
+		MinItems: 1,
+		Required: true,
+		Elem: &schema.Schema{
+			Type:         schema.TypeString,
+			ValidateFunc: validation.IsUUID,
+		},
+	}
+
+	r.Schema["instructions"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Optional: true,
+	}
+
+	r.Schema["requester_can_approve"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+		Default:  false,
+	}
+
+	r.Schema["minimum_required_approvers"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Optional: true,
+	}
+
+	r.Schema["timeout"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Optional: true,
+		Computed: true,
+	}
+
+	return r
+}
+
+func flattenCheckApproval(d *schema.ResourceData, check *pipelineschecksextras.CheckConfiguration, projectID string) error {
+	err := doBaseFlattening(d, check, projectID)
+	if err != nil {
+		return err
+	}
+
+	if check.Settings == nil {
+		return fmt.Errorf("settings nil")
+	}
+
+	check.Type = approvalAndCheckType.Approval
+
+	settings := check.Settings.(map[string]interface{})
+
+	if instructions, found := settings["instructions"]; found {
+		d.Set("instructions", instructions)
+	} else {
+		return fmt.Errorf("instructions not found")
+	}
+
+	if minRequiredApprovers, found := settings["minRequiredApprovers"]; found {
+		d.Set("minimum_required_approvers", minRequiredApprovers)
+	} else {
+		return fmt.Errorf("minRequiredApprovers not found")
+	}
+
+	if requesterCannotBeApprover, found := settings["requesterCannotBeApprover"]; found {
+		d.Set("requester_can_approve", !requesterCannotBeApprover.(bool))
+	} else {
+		d.Set("requester_can_approve", true)
+	}
```

Comment:
`requester_can_approve`  defaults to `false`, no need to set to `true` if not returns be service
```suggestion
	if requesterCannotBeApprover, found := settings["requesterCannotBeApprover"]; found {
		d.Set("requester_can_approve", !requesterCannotBeApprover.(bool))
	} 
```